docs/source/extensions_tutorial/6_documenting_extension.rst

```diff
@@ -0,0 +1,82 @@
+Documenting Extensions
+----------------------
+
+Using the same tools used to generate the documentation for the `NWB-N core format <https://nwb-schema.readthedocs.io/en/latest/>`_
+one can easily generate documentation in HTML, PDF, ePub and many other format for extensions as well.
+
+Code to generate this documentation is maintained in a separate repo: https://github.com/hdmf-dev/hdmf-docutils. To use these utilities, install the package with pip:
+
+.. code-block:: text
+
+    pip install hdmf-docutils
+
+For the purpose of this example, we assume that our current directory has the following structure.
+
+
+.. code-block:: text
+
+    - my_extension/
+      - my_extension_source/
+          - mylab.namespace.yaml
+          - mylab.specs.yaml
+          - ...
+          - docs/  (Optional)
+              - mylab_description.rst
+              - mylab_release_notes.rst
+
+In addition to Python 3.x, you will also need ``sphinx`` (including the ``sphinx-quickstart`` tool) installed.
+Sphinx is available here http://www.sphinx-doc.org/en/stable/install.html .
+
+We can now create the sources of our documentation as follows:
+
+.. code-block:: text
+
+    python3 nwb_init_sphinx_extension_doc  \
+                 --project test \
+                 --author "Dr. Master Expert" \
+                 --version "1.2.3" \
+                 --release alpha \
+                 --output my_extension_docs \
+                 --spec_dir my_extension_source \
+                 --namespace_filename mylab.namespace.yaml \
+                 --default_namespace mylab
+                 --external_description my_extension_source/docs/mylab_description.rst \  (Optional)
+                 --external_release_notes my_extension_source/docs/mylab_release_notes.rst \  (Optional)
```

Comment:
I believe the NDX-template already sets all this up for you. As such, I would suggest to focus here on customizing the documentation and setup generated by the NDX template. This part could then be a separate section on how to create your own documentation from scratch, but I'm not sure this is necessary here or maybe that should go to the hdmf-docutils and simply linked from here.